docs: add create custom theme tutorial (#593)
Refinitiv/refinitiv-ui: documents/src/pages/build-app/custom-theme.md

| src/colors.less               | Colour palette to use in your custom theme                                                |
| index.less                    | Less entry point which import base theme, colour palette and variables from each variants |

## Customise theme global variables

To customise styles for your theme, it is often done by overriding value of less global variables. For example, to change primary color for your dark theme, you can override value of `@scheme-color-primary` inside `./variants/dark/variables.less`.

In the sample custom theme project, it overrides theme primary colour, tertiary color and some colours of button for dark theme.

```less
@scheme-color-primary: @color-seafoam-400;
@scheme-color-tertiary: @color-gray-600;
@button-hover-background-color: @scheme-color-primary;
@button-pressed-background-color: @scheme-color-primary;
```

List of all less variables that you could override are in [Elemental Theme](https://github.com/Refinitiv/refinitiv-ui/blob/v6/packages/elemental-theme/src/variables.less) and [Halo Theme](https://github.com/Refinitiv/refinitiv-ui/blob/v6/packages/halo-theme/src/variants/dark/variables.less).

> @> Elemental Theme is super base theme of every themes in Element Framework, including Halo Theme. With the same concept as creating custom theme, Halo Theme is created by extending from Elemental Theme and overriding less variables, [see example](https://github.com/Refinitiv/refinitiv-ui/blob/v6/packages/halo-theme/src/variants/dark/overrides.less).

Reviewer: ```suggestion
@> Elemental theme is a base theme of every themes in Element Framework, including Halo theme. Similarly, the same concept applies when creating custom theme, Halo theme is created by extending from Elemental theme and overriding less variables, [see example](https://github.com/Refinitiv/refinitiv-ui/blob/v6/packages/halo-theme/src/variants/dark/overrides.less).
```

## Customise theme native styles

> Some styles are required to be applied in native HTML elements such as font-family. Those styles can be set in less files in `native-elements` folder. For example, in the custom theme sample project, font-family and padding are customised for body HTML tag.

Reviewer: ```suggestion
Some styles are required to be applied into native HTML elements such as `font-family` to `body`. To so, you can add less file with name same as HTML tag to `native-elements` folder.

For example, in the custom theme sample project, `font-family` and `padding` are customised for body HTML tag.
```

```less
// import native component style from base theme
@import "@refinitiv-ui/halo-theme/src/native-elements/body";

// Customise native component style
body {
  padding: 20px;
  font-family: sans-serif;
}
```

## Customise component styles

> When styles of components are needed to be customised, you can customise styles of an individual Element Framework component by adding less file(s) in `custom-elements` folder. Custom theme sample project shows how you can customise styles of host and an internal part of `ef-toggle`.

Reviewer: ```suggestion
To customise a component styles, add `less` file with the same name as component's name into `custom-elements` folder. 

Provided sample project shows how you can customise styles of host and an internal part of `ef-toggle`.
```

```less
:host {
  border-radius: 10px;

  [part="toggle"] {
    &::after {
      border-radius: 50%;
    }
  }
}
```

> You only need to create less file for components that you need to customise. Any components that do not need any customisation, theme compiler will automatically extends it directly from Halo Theme and output the files in `./dark` and `./light` folder.

Reviewer: ```suggestion
You only need to create less file for components that you need to customise. The rest of components that do not need further customisation will be automatically extended from Halo theme.
```
